Mathlib/Analysis/Convex/Cone/Proper.lean

```diff
@@ -69,62 +68,96 @@ theorem closure_eq {K L : ConvexCone 𝕜 E} : K.closure = L ↔ closure (K : Se
 
 end ConvexCone
 
-/-- A proper cone is a convex cone `K` that is nonempty and closed. Proper cones have the nice
-property that the dual of the dual of a proper cone is itself. This makes them useful for defining
-cone programs and proving duality theorems. -/
+namespace PointedCone
+
+variable {𝕜 : Type*} [OrderedSemiring 𝕜]
+
+variable {E : Type*} [AddCommMonoid E] [TopologicalSpace E] [ContinuousAdd E] [Module 𝕜 E]
+  [ContinuousConstSMul 𝕜 E]
+
+lemma closure_aux (K : PointedCone 𝕜 E) : (K : ConvexCone 𝕜 E).closure.Pointed :=
+  subset_closure $ PointedCone.toConvexCone_pointed _
+
+/-- The closure of a pointed cone inside a topological space as a pointed cone. This
+construction is mainly used for defining maps between proper cones. -/
+protected def closure (K : PointedCone 𝕜 E) : PointedCone 𝕜 E :=
+  ConvexCone.toPointedCone K.closure_aux
+
+@[simp, norm_cast]
+theorem coe_closure (K : PointedCone 𝕜 E) : (K.closure : Set E) = closure K :=
+  rfl
+
+@[simp]
+protected theorem mem_closure {K : PointedCone 𝕜 E} {a : E} :
+    a ∈ K.closure ↔ a ∈ closure (K : Set E) :=
+  Iff.rfl
+
+@[simp]
+theorem closure_eq {K L : PointedCone 𝕜 E} : K.closure = L ↔ closure (K : Set E) = L :=
+  SetLike.ext'_iff
+
+end PointedCone
```

Comment:
Is there a reason these results don't live in the `Pointed` file?

Reply:
It's because of the import hierarchy. I would need to move `ConvexCone.closure` to the `ConvexCone/Basic.lean` file but then import the required topological spaces stuff there. Right now the only import in `ConvexCone/Basic.lean` is `import Mathlib.Analysis.Convex.Hull`.

Reply:
Should I move both `ConvexCone.closure` and `PointedCone.closure` to a separate file?